lib/rgeo/cartesian/feature_methods.rb

```diff
@@ -50,31 +50,161 @@ def segments
       end
 
       def is_simple?
-        len = segments.length
-        return false if segments.any?(&:degenerate?)
-        return true if len == 1
-        return segments[0].s != segments[1].e if len == 2
-        segments.each_with_index do |seg, index|
-          nindex = index + 1
-          nindex = nil if nindex == len
-          return false if nindex && seg.contains_point?(segments[nindex].e)
-          pindex = index - 1
-          pindex = nil if pindex < 0
-          return false if pindex && seg.contains_point?(segments[pindex].s)
-          next unless nindex
-          oindex = nindex + 1
-          while oindex < len
-            oseg = segments[oindex]
-            return false if !(index == 0 && oindex == len - 1 && seg.s == oseg.e) && seg.intersects_segment?(oseg)
-            oindex += 1
-          end
-        end
-        true
+        # TODO: should we replace with graph.incident_edges.length == segments.length
+        # since graph isn't used elsewhere yet it might be more overhead.
```

Comment:
Not 100% sure I get your point there ^^

Reply:
The same simplicity check (self-intersection) can be performed by creating the `GeometryGraph` of the `LineString` and doing the described comparison. But doing the check with `graph` will be slower since it has to perform an intersection check and then link all of the nodes whereas the current implementation only does an intersection check.

And since `graph` is not used elsewhere in any `LineString` method I figured the additional overhead is not necessary.